spec/controllers/api/teams_controller_spec.rb

```diff
@@ -7,80 +7,185 @@
 
-      expect(attributes['name']).to eq team.name
-      expect(attributes['description']).to eq team.description
+      expect(included.size).to be(2)
+      expect(folder_relationships_length).to be(1)
     end
 
-    it 'should get multiple teams if more than one id is given' do
+    it 'should get team for specific team name' do
+      add_bob_to_team(team3, team3_user)
       login_as(:bob)
 
-      team = teams(:team1)
-      team2 = teams(:team2)
+      get :index, params: { 'q': team3.name }, xhr: true
+
+      expect(data.count).to eq(1)
+      expect(response.status).to be(200)
+
+      attributes_team = data.first['attributes']
+
+      expect(team3.attributes).to include(attributes_team)
+      folder_relationships_length = data.first['relationships']['folders']['data'].size
+
+      expect(included.size).to be(2)
+      expect(folder_relationships_length).to be(1)
+    end
+
+    it 'should get folder for specific folder name' do
+      add_bob_to_team(team3, team3_user)
+      login_as(:bob)
 
-      get :index, params: { 'team_ids': [team.id, team2.id].join(',') }, xhr: true
+      folder = team3.folders.first
 
-      expect(data.count).to eq(2)
+      get :index, params: { 'q': folder.name }, xhr: true
+
+      expect(data.count).to eq(1)
       expect(response.status).to be(200)
 
-      attributes_first_team = data.first['attributes']
-      attributes_second_team = data.second['attributes']
+      relationships_folder_team = data.first['relationships']['folders']['data'].first
 
-      included_types = json['included'].map { |e| e['type'] }
+      expect(relationships_folder_team['id'].to_i).to eq folder.id
+      expect(relationships_folder_team['type']).to eq 'folders'
+      folder_relationships_length = data.first['relationships']['folders']['data'].size
 
-      expect(included_types).to include('folder'.pluralize)
-      expect(included_types).to include('account'.pluralize)
+      expect(included.size).to be(2)
+      expect(folder_relationships_length).to be(1)
+    end
 
-      expect(attributes_first_team['name']).to eq team.name
-      expect(attributes_first_team['description']).to eq team.description
+    it 'should get account for specific account name' do
+      add_bob_to_team(team3, team3_user)
+      login_as(:bob)
+
+      folder = team3.folders.first
+      account = folder.accounts.first
 
-      expect(attributes_second_team['name']).to eq team2.name
-      expect(attributes_second_team['description']).to eq team2.description
+      get :index, params: { 'q': account.accountname }, xhr: true
+
+      expect(data.count).to eq(1)
+      expect(response.status).to be(200)
+
+      included_account = included.second
+
+      expect(included_account['id'].to_i).to eq account.id
+      expect(included_account['attributes']['accountname']).to eq account.accountname
+      expect(included_account['attributes']['description']).to eq account.description
```

Comment:
add test case for filter account description !
it should also match terms in description